src/main/java/me/botsko/prism/listeners/PrismVehicleEvents.java

```diff
@@ -21,132 +17,91 @@
+
+        final Vehicle vehicle = event.getVehicle();
+        final Location loc = vehicle.getLocation();
+
+        final String coord_key = loc.getBlockX() + ":" + loc.getBlockY() + ":" + loc.getBlockZ();
+        String value = plugin.preplannedVehiclePlacement.get(coord_key);
+        UUID uuid = null;
+        try {
+            uuid = UUID.fromString(value);
+        } catch (Exception ignored) {
+        }
+        final OfflinePlayer player = uuid != null ? Bukkit.getOfflinePlayer(uuid) : null;
+        if (player != null) {
+            // TODO: name ref
+            if (!Prism.getIgnore().event("vehicle-place", loc.getWorld(), player.getName()))
+                return;
+            RecordingQueue.addToQueue(ActionFactory.createVehicle("vehicle-place", vehicle, player));
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onVehicleDestroy(final VehicleDestroyEvent event) {
+
+        final Vehicle vehicle = event.getVehicle();
+        final Entity attacker = event.getAttacker();
+        // Was it broken by an attack
+        if (attacker != null) {
+            handlePlayerAction(attacker,vehicle,"vehicle-break");
+        } else {
+
+            // Otherwise its driver was reckless
+            final List<Entity> passengers = vehicle.getPassengers();
+            if (!passengers.isEmpty()) {
+                Entity passenger = passengers.get(0);
+                handlePlayerAction(passenger,vehicle,"vehicle-break");
+            }
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onVehicleEnter(final VehicleEnterEvent event) {
+        final Vehicle vehicle = event.getVehicle();
+        final Entity entity = event.getEntered();
+        handlePlayerAction(entity,vehicle,"vehicle-enter");
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onVehicleExit(final VehicleExitEvent event) {
+
+        final Vehicle vehicle = event.getVehicle();
+        final Entity entity = event.getExited();
+        handlePlayerAction(entity,vehicle,"vehicle-exit","vehicle-enter");
+    }
+
+    private void handlePlayerAction(Entity entity, Vehicle vehicle, String action){
+        handlePlayerAction(entity,vehicle,action,action);
+    }
+
+    private void handlePlayerAction(Entity entity, Vehicle vehicle, String action, String customCheck){
+        if (entity instanceof Player) {
+            if (!Prism.getIgnore().event(customCheck, ((Player) entity)))
+                return;
+            RecordingQueue.addToQueue(ActionFactory.createVehicle(action, vehicle, (Player) entity));
+        } else {
+            if (!Prism.getIgnore().event(customCheck, entity.getWorld()))
+                return;
+            RecordingQueue.addToQueue(ActionFactory.createVehicle(action, vehicle, entity.getType().name().toLowerCase()));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 123).](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)